### PR TITLE
New way to handle big numbers formatting, export magnitude symbols

### PR DESCRIPTION
**Pull request by mvaivre**

### CODE REVIEW

File: test/numbers-test.ts [hunk 1]
```diff
@@ -135,7 +135,23 @@ it('Should abbreviate token amount', () => {
...
+        amountDecimals: 17
+      })
+    ).toEqual("100'000.00T"),
+    expect(
+      formatAmountForDisplay({
+        amount: BigInt('99999999999999999999999999999999999'),
+        amountDecimals: 17,
+        displayDecimals: 2
+      })
+    ).toEqual("1'000'000.00T"), // TODO: Fix edge case. This should ideally output 999'999.99T
```
Comment (mvaivre) [inline test/numbers-test.ts:151]: @nop33 My brain is a bit fried, but we could either add an option to truncate the result (here it would then output 999'999.99T), or we could truncate by default. Or we keep rounding...

WDYT?

Comment (nop33) [reply test/numbers-test.ts:151]: If I understand everything correctly, since this displays 2 decimals, this means we round on the 3rd decimal which means that this should not ideally output `999'999.99 T` but `1.00e+18`. To output `999'999.99 T` we need this as input (notice the `3`) :

```ts
formatAmountForDisplay({
  amount: BigInt('99999999999999999999399999999999999'),
  amountDecimals: 17,
  displayDecimals: 2
})

```

I'll do some tests...

Comment (nop33) [reply test/numbers-test.ts:153]: @mvaivre shouldn't the algorithm return `1e+18` here instead of `1'000'000.00T`? Didn't we agree that the highest number to display before the `e` notation is `999'999T`?

**Review by nop33 — COMMENTED** [review]
When the last comment is addressed (ensuring that we never run in the case of displaying anything above 999'999 T) then we are ready to merge!
